src/main/groovy/io/saagie/plugin/dataops/clients/SaagieClient.groovy

```diff
@@ -1609,23 +1722,56 @@ class SaagieClient {
 				
 			}
 			
+			def processVariableToImport = { newMappedVariable, variable, id ->
+				
+				def newlyCreatedVariable = saveEnvironmentVariable(newMappedVariable)
+				
+				response.variable << [
+						id   : newlyCreatedVariable.id,
+						name : newlyCreatedVariable.name
+				]
+				
+			}
+			
 			if (jobsConfigFromExportedZip?.jobs) {
 				ImportJobService.importAndCreateJobs(
 						jobsConfigFromExportedZip.jobs,
 						configuration,
-						callbackJobToDebug
+						processJobImportation
 				)
 			}
 			
-			if ( pipelinesConfigFromExportedZip?.pipelines && response.status == 'success') {
+			if (pipelinesConfigFromExportedZip?.pipelines && response.status == 'success') {
 				def newlistJobs = getJobListByNameAndId()
 				ImportPipelineService.importAndCreatePipelines(
 						pipelinesConfigFromExportedZip.pipelines,
 						configuration,
-						callbackPipelinesToDebug,
+						processPipelineToImport,
 						newlistJobs
 				)
 			}
+			
+			if (variablesConfigFromExportedZip?.variables && response.status == 'success') {
+				
+				
+				
+				variablesConfigFromExportedZip?.variables?.values()?.forEach { variable ->
+					def listVariablesByNameAndId = VariableService.instance.getVariableList(variable.configOverride, configuration.project.id, this.&getVariableEnvironmentFromList)
```

Comment:
what is the & in `this.&getVariableEnvironmentFromList`

Reply:
This is property in groovy that will send method reference to pass it as a parameter, then we can execute it inside another methode